### PR TITLE
Add an option to include disabled parts in the XML save file

### PR DESCRIPTION
**Pull request by zynmuse**
This change provides a configuration parameter which when switched on will alter the save mechanism to include disabled zynaddsubfx Parts in the  xmz file. With this implementation the default for the new option id OFF.
A new option is provided on the fltk settings screen.
A further change to mruby-zest build will provide a similar option on the zest settings screen.
 
This should fix zyn-fusion-issues #262.

### CODE REVIEW

**Comment by fundamental** [comment]
There's currently some test failures and some refactoring needed to get this PR merged. The concept is sound though.

File: src/Misc/Part.cpp [hunk 1]
```diff
@@ -1128,7 +1129,7 @@ void Part::add2XML(XMLwrapper& xml)
 {
     //parameters
     xml.addparbool("enabled", Penabled);
-    if((Penabled == 0) && (xml.minimal))
+    if(Penabled == 0 && SaveFullXml == 0 && (xml.minimal))
```
[the remark this answers is not in the document]
Comment (zynmuse) [reply src/Misc/Part.cpp:1151]: Intend to clear xml.minimal before save_parameters gets called and remove SaveFullXml from here

File: src/Misc/Part.cpp [hunk 2]
```diff
@@ -1276,7 +1277,7 @@ void Part::getfromXMLinstrument(XMLwrapper& xml)
             if(xml.enterbranch("INSTRUMENT_KIT_ITEM", i) == 0)
                 continue;
             setkititemstatus(i, xml.getparbool("enabled", kit[i].Penabled));
-            if(kit[i].Penabled == 0) {
+            if(kit[i].Penabled == 0 && SaveFullXml ==0) {
```
[the remark this answers is not in the document]
Comment (zynmuse) [reply src/Misc/Part.cpp:1303]: OK

File: src/Misc/Part.h [hunk 3]
```diff
@@ -227,6 +227,7 @@ class Part
         const SYNTH_T &synth;
         const AbsTime &time;
         const int &gzip_compression, &interpolation;
+        const int &SaveFullXml;
```
[the remark this answers is not in the document]
Comment (zynmuse) [reply src/Misc/Part.h:234]: intend to remove this

File: src/Params/ADnoteParameters.cpp [hunk 4]
```diff
@@ -740,7 +740,7 @@ void ADnoteParameters::add2XMLsection(XMLwrapper& xml, int n)
     }
 
     xml.addparbool("enabled", VoicePar[nvoice].Enabled);
-    if(((VoicePar[nvoice].Enabled == 0) && (oscilused == 0)
+    if((VoicePar[nvoice].Enabled == 0  && SaveFullXml==0 && (oscilused == 0)
```
[the remark this answers is not in the document]
Comment (zynmuse) [reply src/Params/ADnoteParameters.cpp:749]: xml.mimimal with be false and SaveFullXML will be removed from here

File: src/Params/ADnoteParameters.h [hunk 5]
```diff
@@ -336,6 +336,7 @@ class ADnoteParameters:public PresetsArray
 
         const AbsTime *time;
         int64_t last_update_timestamp;
+        int SaveFullXml;
```
[the remark this answers is not in the document]
Comment (zynmuse) [reply src/Params/ADnoteParameters.h:339]: Will remove this

File: src/Misc/Config.cpp [hunk 6]
```diff
@@ -209,6 +210,7 @@ void Config::init()
     cfg.GzipCompression = 3;
 
     cfg.Interpolation = 0;
+    cfg.SaveFullXml = 1;
```
[the remark this answers is not in the document]
Comment (zynmuse) [reply src/Misc/Config.cpp:216]: Agree should default to 0;

**Pull request closed without merging**